applications/nrf5340_audio/src/audio/audio_system.c

```diff
@@ -115,35 +117,68 @@ static void encoder_thread(void *arg1, void *arg2, void *arg3)
 	uint32_t blocks_locked_num;
 
 	int debug_trans_count = 0;
-	size_t encoded_data_size = 0;
 
-	void *tmp_pcm_raw_data[CONFIG_FIFO_FRAME_SPLIT_NUM];
 	char pcm_raw_data[FRAME_SIZE_BYTES];
 
-	static uint8_t *encoded_data;
 	static size_t pcm_block_size;
 	static uint32_t test_tone_finite_pos;
 
 	while (1) {
 		/* Don't start encoding until the stream needing it has started */
 		ret = k_poll(&encoder_evt, 1, K_FOREVER);
+		struct audio_data audio_frame = {0};
+		struct net_buf *frame_buf = net_buf_alloc(&fifo_rx_pool, K_NO_WAIT);
+
+		if (frame_buf == NULL) {
+			LOG_ERR("Out of RX buffers");
+			continue;
+		}
```

Comment:
If this happens, then the encoder event will have been polled, but is there any data that needs to be discarded or anything like that?

Reply:
No, the polling event just ensures that the encoder thread sleeps if there is no need for encoding. We could consider adding a purge of the `fifo_rx_pool` when `audio_system_encoder_stop()` is called though.